src/pose_estimating.h

```diff
@@ -0,0 +1,162 @@
+    ConnectedJoint head;
+    ConnectedJoint neck;
+    ConnectedJoint shoulder;
+    ConnectedJoint hip;
+    ConnectedJoint knee;
+    ConnectedJoint foot;
+}
+
+/*
+ * @brief This class handles representations of the user's pose and calculates any updates to their pose that is required.
+ * 
+ * This class stores representations of the users current pose and their calibrated ideal pose, from the results of the PostProcessor class.
+ *
+ * This class handles:
+ * Updatin current user's pose or ideal user's pose
+ * Calculating the angle between ConnectedJoints
+ * Comparing the current user's pose and ideal user's pose
+ * 
+ */
+class PoseEstimater {
+ public:
+     /*
+      * @brief The user's current pose from most recent data provided
+     */
+     Pose currentPose;
+     /* 
+      * @brief The user's ideal pose, expected to be generated during user calibration  
+      */
+     Pose idealPose;
+     /*
+      * @brief Returns the user's current pose
+      * 
+      * @returns Pose
+      */
+     Pose get_current_pose();
+     /*
+      * @brief Updates the user's current pose from PostProcesser results
+      *
+      * @param results ProcessedResults struct containing user's pose data
+      */
+     void updateCurrentPose(PostProcessing :: ProcessedResults results);
+     /*
+      * @brief Returns the user's ideal pose
+      * 
+      * @returns Pose
+      */
+     Pose get_ideal_pose();
+     /*
+      * @brief Updates the user's ideal pose from PostProcesser results
+      *
+      * @param results ProcessedResults struct containing user's pose data
+      */
+     void updateIdealPose(PostProcessing :: ProcessedResults results);
+     /*
+      * @brief Compares user's current pose and ideal pose, returing a pose object, with the angles of the ConnectJoints representing the change needed for each one
+      *
+      * @returns Pose
+      */  
+     Pose calculatePoseChangeNeeded();
+
+ private:
+     /*
+      * @brief Calculates the angle(in degrees) between two points, anticlockwise from origin / bottom left corner of image
+      * @param x1 Relative X co-ordinate of first point
+      * @param y1 Relative Y co-ordinate of second point
+      * @param x2 Relative X co-ordinate of first point
+      * @param y2 Relative Y co-ordinate of first point
+      *
+      * @returns float
+      */
+     float getLineAngle(float x1, float y1, float x2, float y2)
+     /*
+      * @brief Converts `PostProcesser::ProcessedResults` to a `Pose::Pose`
+      *
+      * @param results  The PostProcessor ::ProceccsedResults struct from PostProcessor::PostProcessor being run
+      * @return JointsPose
+      */
+     Pose createPose(PostProcessing :: ProcessedResults results);
+     /*
+      * @brief sets user's current pose
+      *
+      * @param Pose
+      */
+     void set_current_pose(Pose current);
+     /*
+      * @brief sets user's ideal pose
+      *
+      * @param Pose
+      */
+     void set_idea_pose(Pose current);
```

Comment:
Are these needed? If they are private setters it may be worth hiding this functions as implementation details (if they are needed).

Reply:
Well, I was unsure as while they are generally private if someone wants to modify the code in the future I feel its better to document everything?

Reply:
Sounds good

Reply:
I changed my mind. I don't think there's any point